model/src/main/kotlin/ScannerRun.kt

```diff
@@ -202,3 +265,36 @@ private fun scanResultForProvenanceResolutionIssue(packageProvenance: KnownProve
 
 private fun ScanSummary.addIssue(issue: Issue?): ScanSummary =
     if (issue == null) this else copy(issues = (issues + issue).distinct())
+
+private fun mergeFileLists(fileListByPath: Map<String, FileList>): FileList {
+    require(fileListByPath.containsKey("")) {
```

Comment:
For maps, `containsKey` is the same as `contains`, and so the `in` operator can be used here: `"" in fileListByPath`.

Reply:
Also, what's the semantics of `""` as a key? Is it the root path? Does it maybe make sense to define something like `ROOT_PATH_KEY = ""` for this for clarity?

Reply:
> Also, what's the semantics of `""` as a key? Is it the root path? Does it maybe make sense to define something like `ROOT_PATH_KEY = ""` for this for clarity?

I believe that the error message for the require block clarifies that. Agree?

Reply:
The require block still just says "There must be a file list associated with the empty path", which doe snot explain what it *means* to associate a file list with the empty path. Maybe instead: "There must be a file list associated with the root path" (if that's the correct semantics)?